js/search.js

```diff
@@ -0,0 +1,151 @@
+let lastSearch = '';
+
+class Search {
+  constructor(containerDom, resultsSearch) {
+
+    this.container = containerDom;
+    this.resultsSearch = resultsSearch;
+    this.generate();
+  }
+
+
+  async getData(stockSymbol) {
+    const searchUrl = `https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/search?query=${stockSymbol}&limit=10&exchange=NASDAQ`;
+    try {
+      const data = await getJson(searchUrl);
+
+      if (!data || !Array.isArray(data) || !data.length) {
+        return console.warn("Not Found Result - Data Array");
+      }
+
+      lastSearch = stockSymbol;
+      return data;
+
+    } catch {
+      console.error('It was a problem');
+    }
+  }
+
+  refillFeeder(isLoading) {
+    const searchButton = this.searchButton;
+
+    if (isLoading) {
+      searchButton.disabled = true;
+      searchButton.innerHTML = '';
+      for (let i = 0; i < 5; i++) {
+        const loaderDots = document.createElement('span');
+        searchButton.append(loaderDots);
+      }
+    } else {
+      searchButton.innerHTML = 'Search';
+      searchButton.disabled = false;
+    }
+  }
+
+  async handlerSearch() {
+    this.refillFeeder(true);
+
+    const input = this.inputField.value;
+
+    this.resultsSearch.clearAll();
+
+    const data = await this.getData(input);
+
+    if (data) await this.resultsSearch.addAllRes(data);
+
+    else this.resultsSearch.addNothingFound();
+    this.refillFeeder(false);
+  }
+
+  userEnterSearch(event) {
+    if (event.key === 'Enter') {
+      this.handlerSearch();
+    }
+  }
+
+
+
+  generate() {
+    this.searchButton = creatElement('button', {
+      className: 'searchBtn loader-jump',
+      innerText: 'Search',
+    });
+    this.inputField = creatElement('input', {
+      className: 'input-field',
+      placeholder: 'Search for company stock symbol',
+    });
+
+    this.container.append(this.inputField, this.searchButton);
+    this.searchButton.addEventListener('click', e => this.handlerSearch(e));
+    this.inputField.addEventListener('keydown', e => this.userEnterSearch(e));
+  }
+};
+
+
+
+
+
+class ResultsSearch {
```

Comment:
It's better to have seperate file for each class. this is a convention that make your code easier to read and understand.